### PR TITLE
(feat) O3-3392: Patient registration fails to report field errors while saving new patient

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary
This updates error feedback if an error occurs when registering a patient. 

## Screenshots
https://github.com/openmrs/openmrs-esm-patient-management/assets/28008754/b3fe19a4-c8bb-43c9-9b25-431f9d50c474

### CODE REVIEW

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx [hunk 1]
```diff
@@ -106,27 +104,16 @@ export const PatientRegistration: React.FC<PatientRegistrationProps> = ({ savePa
...
-        });
-      } else {
-        createErrorHandler()(error);
-      }
+      const errorMessages = extractErrorMessagesFromResponse(error.responseBody);
+
+      showSnackbar({
+        title: inEditMode
+          ? t('updatePatientErrorSnackbarTitle', 'Patient Details Update Failed')
+          : t('registrationErrorSnackbarTitle', 'Patient Registration Failed'),
```
Comment (denniskigen) [inline packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx:112]:
```suggestion
          ? t('updatePatientErrorSnackbarTitle', 'Patient details update failed')
          : t('registrationErrorSnackbarTitle', 'Patient registration failed'),
```

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx [hunk 2]
```diff
@@ -357,7 +359,14 @@ describe('Registering a new patient', () => {
     expect(mockedSavePatient).toHaveBeenCalledTimes(1);
     expect(mockedSaveEncounter).toHaveBeenCalledTimes(1);
 
-    expect(mockedShowSnackbar).toHaveBeenCalledWith(expect.objectContaining({ subtitle: 'an error message' })),
+    expect(mockedShowSnackbar).toHaveBeenCalledWith(
+      expect.objectContaining({
+        isLowContrast: true,
+        kind: 'error',
+        subtitle: 'an error message',
+        title: 'Patient Registration Failed',
```
Comment (denniskigen) [inline packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx:367]:
```suggestion
        title: 'Patient registration failed',
```

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx [hunk 3]
```diff
@@ -366,6 +375,82 @@ describe('Registering a new patient', () => {
...
+    mockedSavePatient.mockRejectedValueOnce(mockFieldErrorResponse);
+
+    const registerButton = screen.getByRole('button', { name: /Register patient/i });
+    await user.click(registerButton);
+
+    expect(mockedShowSnackbar).toHaveBeenCalledWith({
+      isLowContrast: true,
+      kind: 'error',
+      subtitle: 'Identifier 1245647 already in use by another patient',
+      title: 'Patient Registration Failed',
```
Comment (denniskigen) [inline packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx:451]:
```suggestion
      title: 'Patient registration failed',
```

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @donaldkibet. So this is reproducible by registering a patient with an apostrophe in their name. Is that a known issue in O2 as well? Also, do you know why the error response returns duplicate error messages, @ibacher? We should probably only show one if they're guaranteed to be duplicated.

```tsx
{
    "error": {
        "message": "Invalid Submission",
        "code": "webservices.rest.error.invalid.submission",
        "globalErrors": [],
        "fieldErrors": {
            "names[0].middleName": [
                {
                    "code": "MiddleName.invalid",
                    "message": "MiddleName.invalid"
                },
                {
                    "code": "MiddleName.invalid",
                    "message": "MiddleName.invalid"
                }
            ]
        }
    }
}
```

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx (hunk 2 above)
Comment (denniskigen) [reply packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx:367]: Prefer sentence case.

**Comment by brandones** [comment]
> Only because someone set a GP on dev3 that should not be set in general.

Thank goodness we have people always messing with things they shouldn't be, so that we have our systems chaos-tested

SUDT: screw-up driven testing

**Comment by denniskigen** [comment]
Which GP is that, @ibacher?

**Comment by ibacher** [comment]
Something like `patient.nameValidationRegex`. I've unset it for dev3.

**Comment by ojwanganto** [comment]
That GP usually has a default value though. It is worth checking.
@donaldkibet I guess we still need to ability to inform users about the errors. I can see you have closed the PR.

**Comment by ibacher** [comment]
> That GP usually has a default value though.

In [the platform](https://github.com/openmrs/openmrs-core/blob/8d3e65b48ea3d41c1270c609a081b363849c8c41/api/src/main/resources/org/openmrs/liquibase/snapshots/core-data/liquibase-core-data-2.6.x.xml#L3104), it doesn't, at least not anymore.

**Pull request closed without merging**